### DISCUSSION TITLE
🌟💡 YOLOv5 Study: Hybrid Autolabelling

### DISCUSSION THREAD

**Discussion by glenn-jocher** [ultralytics/yolov5#2379] [Ideas]
## Study 🤔

I had an idea to hybrid label the COCO train set, and then retrain on it. This discussion is mainly a placeholder for later results on this. 

In the process of this I realized there is no easy way to run test.py on the train set, so I created PR https://github.com/ultralytics/yolov5/pull/2453 to allow for selection of any of the 3 sets defined in a data.yaml file (train, val, test) by passing them to the --task argument. After this PR hybrid autolabeling command is:
```
python test.py --task train --data coco.yaml --weights yolov5x6.pt --img 1280 --iou 0.7 --conf 0.90 --save-txt --save-hybrid --project hybrid --name conf_90
```

## Results 😃

TODO

**Comment by robotaiguy** [comment]
I'm planning on doing similar, because a while back I trained a model with images from both COCO2017 and Open images, by using Fiftyone to download ONLY images with Person labels, and I noticed that there exists plenty of opportunity for helping those datasets, especially Open Images. So I'll be using the biggest, baddest, Yolov5x6 model an inferencing them all with --save-txt enabled.
I should be able to knock it out pretty quick with a 3090 and an A100. I was actually using Yolov5 for hybrid autolabeling this morning.
<img width="1274" alt="Screen Shot 2022-06-29 at 7 40 05 PM" src="https://user-images.githubusercontent.com/60989737/176568790-12989eda-1444-4e93-ac6f-b4e2219f5580.png">

**Reply by glenn-jocher** [reply]
@robotwhispering yes hybrid autolabelling is the best way to improve an existing dataset without touching the existing labels.

**Comment by PraveenMNaik** [comment]
@robotwhispering Sir, pls show us how to configure for autolabelling

**Reply by glenn-jocher** [reply]
@PraveenMNaik you can autolabel image with detect.py or val.py:

```
python detect.py --save-txt 
python val.py --data DATA.yaml --save-txt
```

**Reply by spacewalk01** [reply]
May refer to this repo for autolabeling: https://github.com/cnyvfang/labelGo-Yolov5AutoLabelImg
